services/app-web/src/common-code/healthPlanSubmissionHelpers/provisions.ts

```diff
@@ -0,0 +1,148 @@
+import {
+    ModifiedProvisionsAmendmentRecord,
+    ModifiedProvisionsBaseContractRecord,
+    ModifiedProvisionsCHIPRecord,
+} from '../../constants/modifiedProvisions'
+
+import type { HealthPlanFormDataType } from '../healthPlanFormDataType'
+import {
+    CHIPProvisionType,
+    MedicaidBaseProvisionType,
+    MedicaidAmendmentProvisionType,
+    provisionCHIPKeys,
+    modifiedProvisionMedicaidBaseKeys,
+    modifiedProvisionMedicaidAmendmentKeys,
+    GeneralizedProvisionType,
+    isCHIPProvision,
+    isMedicaidAmendmentProvision,
+    isMedicaidBaseProvision,
+} from '../healthPlanFormDataType/ModifiedProvisions'
+import {
+    isBaseContract,
+    isCHIPOnly,
+    isContractAmendment,
+    isContractWithProvisions,
+} from '../healthPlanFormDataType/healthPlanFormData'
+
+/*
+    Each provision key represents a Yes/No question asked on Contract Details. 
+    This is a set of helper functions that each take in a submission and return provisions related data.
+
+    There are currently three distrinct variants of the provisions: 
+    1. For CHIP amendment
+    2. For non CHIP base contract
+    3. For non CHIP contract amendment
+
+    See also ModifiedProvisions.ts
+*/
+
+// Returns the list of provision keys that apply for given submission variant
+const generateApplicableProvisionsList = (
+    draftSubmission: HealthPlanFormDataType
+):
+    | CHIPProvisionType[]
```

Comment:
I wonder if this type could be used where we're using GeneralizedProvisionType? 

That's my only thought about this PR, the `as unknown as ...` bits don't seem great and I wonder if there's a better way to do them. I'll try and mess with it a little tomorrow and see.